st2common/st2common/util/output_schema.py

```diff
@@ -69,8 +69,9 @@ def mask_secret_output(ac_ex, output_value):
         return output_value
 
     for key, spec in output_schema.items():
-        if key in output_value[output_key] and spec.get("secret", False):
-            output_value[output_key][key] = MASKED_ATTRIBUTE_VALUE
+        if not isinstance(output_value[output_key], bool):
```

Comment:
What about this:

Add `from collections.abc import Iterable` and:
```suggestion
        if isinstance(output_value[output_key], Iterable):
```